### ISSUE TITLE
'iotedge check' can not detect moby-engine in Yocto build 

### ISSUE THREAD

**Issue by Lexmark-chad** [Azure/iotedge#3921]
## Expected Behavior
Check for moby-engine should have passed.

## Current Behavior
`iotedge check` warned that the device was not using a production-supported container engine.

## Steps to Reproduce

We are using the Zeus version of Yocto / Poky.

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```

Configuration checks
--------------------
√ config.yaml is well-formed - OK
√ config.yaml has well-formed connection string - OK
√ container engine is installed and functional - OK
√ config.yaml has correct hostname - OK
× config.yaml has correct URIs for daemon mgmt endpoint - Error
    One or more errors occurred. (Permission denied /var/run/iotedge/mgmt.sock)
‼ latest security daemon - Warning
    Installed IoT Edge daemon has version 1.0.10 but 1.0.10.1 is the latest stable version available.
    Please see https://aka.ms/iotedge-update-runtime for update instructions.
√ host time is close to real time - OK
√ container time is close to host time - OK
‼ DNS server - Warning
    Container engine is not configured with DNS server setting, which may impact connectivity to IoT Hub.
    Please see https://aka.ms/iotedge-prod-checklist-dns for best practices.
    You can ignore this warning if you are setting DNS server per module in the Edge deployment.
√ production readiness: identity certificates expiry - OK
‼ production readiness: certificates - Warning
    The Edge device is using self-signed automatically-generated development certificates.
    They will expire in 81 days (at 2021-01-26 21:22:50 UTC) causing module-to-module and downstream device communication to fail on an active deployment.
    After the certs have expired, restarting the IoT Edge daemon will trigger it to generate new development certs.
    Please consider using production certificates instead. See https://aka.ms/iotedge-prod-checklist-certs for best practices.
‼ production readiness: container engine - Warning
    Device is not using a production-supported container engine (moby-engine).
    Please see https://aka.ms/iotedge-prod-checklist-moby for details.
‼ production readiness: logs policy - Warning
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.
‼ production readiness: Edge Agent's storage directory is persisted on the host filesystem - Warning
    The edgeAgent module is not configured to persist its /tmp/edgeAgent directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.
× production readiness: Edge Hub's storage directory is persisted on the host filesystem - Error
    Could not check current state of edgeHub container

Connectivity checks
-------------------
√ host can connect to and perform TLS handshake with DPS endpoint - OK
√ host can connect to and perform TLS handshake with IoT Hub AMQP port - OK
√ host can connect to and perform TLS handshake with IoT Hub HTTPS / WebSockets port - OK
√ host can connect to and perform TLS handshake with IoT Hub MQTT port - OK
√ container on the default network can connect to IoT Hub AMQP port - OK
√ container on the default network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the default network can connect to IoT Hub MQTT port - OK
√ container on the IoT Edge module network can connect to IoT Hub AMQP port - OK
√ container on the IoT Edge module network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the IoT Edge module network can connect to IoT Hub MQTT port - OK

17 check(s) succeeded.
6 check(s) raised warnings. Re-run with --verbose for more details.
2 check(s) raised errors. Re-run with --verbose for more details.

```
</details>

### Device Information
* Host OS [e.g. Ubuntu 16.04, Ubuntu 18.04, Windows IoT Core]: arm-poky-linux-musleabi
* Architecture [e.g. amd64, arm32, arm64]: arm64
* Container OS [e.g. Linux containers, Windows containers]: Linux containers

### Runtime Versions
* iotedged [run `iotedge version`]:  1.0.10
* Edge Agent [image tag (e.g. 1.0.0)]:  1.0.9.4
* Edge Hub [image tag (e.g. 1.0.0)]:  1.0.9.4
* Docker/Moby [run `docker version`]:

```

Client:
 Version:           19.03.2
 API version:       1.40
 Go version:        go1.12.9
 Git commit:        2f1931f9eb
 Built:             Fri Nov  6 17:25:42 2020
 OS/Arch:           linux/arm64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.2
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.12.9
  Git commit:       08bc39c8f9
  Built:            Fri Nov  6 17:25:06 2020
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          v1.2.7-11-gfd103cb71.m
  GitCommit:        fd103cb716352c7e19768e4fed057f71d68902a0.m
 runc:
  Version:          1.0.0-rc8
  GitCommit:        425e105d5a03fabd737a126ad93d62a9eeede87f-dirty
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683-dirty

``` 

## Logs
N/A

## Additional Information
We are working through the IoT edge production checklist for an embedded product.  The check in iotedge fails here:

https://github.com/Azure/iotedge/blob/release/1.0.10/edgelet/iotedge/src/check/checks/container_engine_is_moby.rs#L81

Our build uses this layer to select and build docker-moby:

https://git.yoctoproject.org/cgit/cgit.cgi/meta-virtualization/?h=zeus

We can get the check to pass by setting DOCKER_VERSION in the docker-moby.bb recipe to include a "+azure" suffix that the check looks for.  This obviously is a hack though.  MSFT's fork of moby is private.  Is there any reason why it is not public?  Are there specific customizations being made to docker that necessitates the repo being private?  I hope that someday soon Microsoft will make Yocto a tier-1 supported environment and when that day comes this issue will need to be addressed.

**Comment by veyalla** [comment]
This is expected as the source for moby-engine supplied by Microsoft is not available publicly. We will provide official guidance on this topic if and when Yocto is supported as a [Tier 1 OS](https://docs.microsoft.com/en-us/azure/iot-edge/support#tier-1) for IoT Edge.

**Comment by micahl** [comment]
Reopening this issue to track a fix to iotedge check.  Note this doesn't imply yocto has been moved to Tier 1.  What has changed is what we do with Moby.  Microsoft's internal fork may have originally contained separate changes, but we've moved away from that. Any fixes / changes that are needed go into the upstream. The internal fork Microsoft maintains is then simply used to create our own packages on our own timeline.
